storage/innobase/log/log0log.cc

```diff
@@ -885,9 +867,8 @@ log_group_file_header_flush(
 		  ? LOG_HEADER_FORMAT_10_3
 		  : LOG_HEADER_FORMAT_10_2));
 
-	buf = *(group->file_header_bufs + nth_file);
+	byte buf[OS_FILE_LOG_BLOCK_SIZE] = {0};
```

Comment:
This is replacing aligned memory with unaligned one. I wonder if the alignment is safe to remove.
I would tend to guess so, because `OS_FILE_LOG_BLOCK_SIZE` is only 512 bytes, not 4096 bytes, which I think could be what unbuffered I/O requires..

Reply:
The answer was in `man 2 open`. I restored the alignment. It's great you didn't allow me to break this as pretty strange things could happen.